BACKLOG-18551 - Implemented code for the avro output dialog according to the ux specification.

pentaho-jrice opened the pull request:
@rmansoor , @bmorrise , @ddiroma .  Hello gents.  Please take a look at PR for AvroOutputDialog.

wingman-pentaho commented:
### Build Skipped

:white_check_mark: There was nothing to build. The changes are probably excluded by the build rules. If this is an error, please contact your system administrator.

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl '.,kettle-plugins/formats' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl '.,kettle-plugins/formats' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl '.,kettle-plugins/formats' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  .gitignore
  kettle-plugins/formats/src/main/java/org/pentaho/big/data/kettle/plugins/formats/avro/output/AvroOutputDialog.java
  kettle-plugins/formats/src/main/java/org/pentaho/big/data/kettle/plugins/formats/avro/output/AvroOutputMeta.java
  kettle-plugins/formats/src/main/resources/AO.svg
  kettle-plugins/formats/src/main/resources/org/pentaho/big/data/kettle/plugins/formats/avro/output/messages/messages_en_US.properties
```

Newly Broken Tests:
======================
###### org.pentaho.big.data.kettle.plugins.kafka.KafkaConsumerInputTest.testExecutesWhenDurationIsReached:

:red_circle: `org.mockito.exceptions.verification.junit.ArgumentsAreDifferent`

```java
org.mockito.exceptions.verification.junit.ArgumentsAreDifferent: 

Argument(s) are different! Wanted:
logChannel.logBasic(
    "
------------> Linenr 1------------------------------
Key = key_0
Message = value_0
Topic = pentaho
Partition = 0
Offset = 0
Timestamp = -1

===================="
);
-> at org.pentaho.big.data.kettle.plugins.kafka.KafkaConsumerInputTest.verifyRow(KafkaConsumerInputTest.java:380)
Actual invocation has different arguments:
logChannel.logBasic(
    "Finished processing (I=
```

#### Checkstyle violations

The code submitted is not complying with one or more of our guidelines. To perform these checks locally, [please use this tool](https://github.com/pentaho/pentaho-coding-standards).

###### /home/buildguy/project/kettle-plugins/formats/src/main/java/org/pentaho/big/data/kettle/plugins/formats/avro/output/AvroOutputDialog.java

> :heavy_exclamation_mark: Unused import - org.eclipse.swt.events.SelectionAdapter. at Line: 29 Column: 8

> :heavy_exclamation_mark: Unused import - org.eclipse.swt.events.SelectionEvent. at Line: 30 Column: 8

> :heavy_exclamation_mark: Using the '.*' form of import should be avoided - org.eclipse.swt.widgets.*. at Line: 34

> :heavy_exclamation_mark: Using the '.*' form of import should be avoided - org.pentaho.di.ui.core.widget.*. at Line: 47

> :heavy_exclamation_mark: '(' is not followed by whitespace. at Line: 197 Column: 32

> :heavy_exclamation_mark: ')' is not preceded with whitespace. at Line: 197 Column: 40

> :heavy_exclamation_mark: '(' is not followed by whitespace. at Line: 300 Column: 37

> :heavy_exclamation_mark: ')' is not preceded with whitespace. at Line: 300 Column: 55

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'kettle-plugins/formats' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl 'kettle-plugins/formats' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'kettle-plugins/formats' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  kettle-plugins/formats/src/main/java/org/pentaho/big/data/kettle/plugins/formats/avro/output/AvroOutputDialog.java
```

#### Checkstyle violations

The code submitted is not complying with one or more of our guidelines. To perform these checks locally, [please use this tool](https://github.com/pentaho/pentaho-coding-standards).

###### /home/buildguy/project/kettle-plugins/formats/src/main/java/org/pentaho/big/data/kettle/plugins/formats/avro/output/AvroOutputDialog.java

> :heavy_exclamation_mark: ')' is not preceded with whitespace. at Line: 312 Column: 56

pentaho-jrice commented:
@rmansoor @ddiroma @bmorrise .  Fixed wingman issues.  Added tech debt jira ticket to comment to let future devs know the code was commented on purpose

wingman-pentaho commented:
### Build Completed

:white_check_mark: This pull request has passed all validations.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl '.,kettle-plugins/formats' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl '.,kettle-plugins/formats' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl '.,kettle-plugins/formats' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  .gitignore
  kettle-plugins/formats/src/main/java/org/pentaho/big/data/kettle/plugins/formats/avro/output/AvroOutputDialog.java
  kettle-plugins/formats/src/main/java/org/pentaho/big/data/kettle/plugins/formats/avro/output/AvroOutputMeta.java
  kettle-plugins/formats/src/main/resources/AO.svg
  kettle-plugins/formats/src/main/resources/org/pentaho/big/data/kettle/plugins/formats/avro/output/messages/messages_en_US.properties
```
